### PR TITLE
Master mdl 71641 86

### PR DESCRIPTION
**Pull request by safatshahin**
PR for #86

### CODE REVIEW

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @safatshahin 

Review done:

User test: OK
Code checker: OK

To check:
1. Can we have a look to /tag/manage.php -> questions, should be this who handle the  enable or disable? we can check this tomorrow.
2. The js refactoring should include ES6 migration?
3. The external functions called in deprecated functions.

Thank you.

**Comment by safatshahin** [comment]
Hi @guillogo 
Thanks for reviewing this, i have made the changes according to the review, regarding the points you mentioned:

1.    Can we have a look to /tag/manage.php -> questions, should be this who handle the enable or disable? we can check this tomorrow.  -> -- plugin enable and disable is something related to core which i think self dependent, but the view or the feature enable or disable should be handled by cfg->usertags, i have implemented that in the child class to act with this setting from moodle.
2.     The js refactoring should include ES6 migration? -- i dont think so, as we are just moving the js files to the appropriate location
3.     The external functions called in deprecated functions -- i was kinda thinking about it from different angles, but calling the correct one or not calling the correct one, wont make difference in this case, so removed the call.

Please let me know if you want any other changes.
Thanks

File: question/bank/tagquestion/db/services.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    array(
+        'qbank_tagquestion_submit_tags_form' =>
+            array(
+                'classname' => 'qbank_tagquestion\external\qbank_tagquestion_external',
+                'methodname' => 'submit_tags_form',
+                'description' => 'Update the question tags.',
+                'type' => 'write',
+                'ajax' => true,
+            ),
+);
```
Comment (guillogo) [inline question/bank/tagquestion/db/services.php:38]: In this line I'm getting this in codechecker: `#38: ); Line indented incorrectly; expected at least 4 spaces, found 0`

**Comment by guillogo** [comment]
Hi @safatshahin thank you for your answers, just one more little change and is ready to be merged. Thank you.

**Comment by guillogo** [comment]
Hi @safatshahin approved and merged. Thank you.

**Pull request merged**